lib/swaps/SwapClient.ts

```diff
@@ -228,6 +228,35 @@ abstract class SwapClient extends EventEmitter {
     }
   };
 
+  protected checkLowBalance = (
+    remoteBalance: number,
+    localBalance: number,
+    totalBalance: number,
+    alertThreshold: number,
+    currency: string,
+    emit: Function,
+  ) => {
+    if (localBalance < alertThreshold) {
+      emit('lowTradingBalance', {
```

Comment:
Is there a reason this is taking the `emit` function here as a parameter instead of `this.emit()` with types for the event? I'm thinking the `emit` and `on` functions can be defined in this class (alongside other generic swap client events like `connectionVerified`) rather than in `LndClient` and `ConnextClient` since they seem to be using the same types. Then this can use `this.emit`.

Reply:
Yes actually, I was thinking the same, but if I do the what you said, then I have to define `BaseAlert` fields such as alert `type, message, date` which are being defined in Alerts middleware class.

Reply:
You're avoiding that not by passing around the `emit` function but by defining it as just a `Function` - which loses all type checking entirely.  This could be done by just hard casting whatever object to the desired type, but that still robs us of the benefits of type checking and could lead to bugs down the road.

I started writing up a proposal for how to refactor this but it wound up being easier just to refactor it myself, so I'll push a commit here to be squashed. Basically you can define a `BalanceAlertEvent` type that only contains the fields specific to balance alerts, and then use that on the swap client level while keeping the `BalanceAlert` within the Alerts middleware. This way we can preserve all the type checking.